### PR TITLE
MMP-217, add readme improve old one, minor fix to java docs

### PR DESCRIPTION
**Pull request by damiansima**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+== Modules
+* mule-classloader-model
+* mule-packager
+* mule-deployer
+* mule-maven-plugin
+* mule-artifact-it
+
+
+=== mule-classloader-model
+Contains the abstract model which describe the descriptor files used when generating a repository for heavy weight packages
```
Comment (diegostrubolini) [inline README.adoc]: describe -> describes

File: README.adoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+=== mule-packager
+Contains the Java API that allows you to create the different Mule artifacts:
+
+* mule-application
+* mule-policy
+* mule-domain
+* mule-domain-bundle
+
+It also contains a set of validator to ensure that the artifact can be generated properly
```
Comment (diegostrubolini) [inline README.adoc]: validator -> validators

File: README.adoc [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+=== mule-deployer
+Contains the Java API that allows you to deploy the Mule artifacts to the different environments
+
+=== mule-maven-plugin
+Is the Maven Plugin that provides Maven integration.
+Through it you can package different Mule artifacts and deploy them to both Nexus Servers and the different Anypoint Environments.
+
+
+=== mule-artifact-it
+Contains the integration test for the modules:
```
Comment (diegostrubolini) [inline README.adoc]: test -> tests

File: mule-maven-plugin/README.adoc [hunk 4]
```diff
@@ -36,547 +34,178 @@ Add https://repository.mulesoft.org/nexus/content/repositories/releases[this] re
 
 You can download the JAR file manually from https://repository.mulesoft.org/nexus/content/repositories/releases/org/mule/tools/maven/mule-maven-plugin/2.0[here] or get if from Maven Central Repository.
 
-== A simple example
+== Usage
+The mule-maven-plugin can be use not only to package artifact but also to deploy them.
```
Comment (diegostrubolini) [inline mule-maven-plugin/README.adoc]: use -> used

File: mule-maven-plugin/README.adoc [hunk 5]
```diff
@@ -36,547 +34,178 @@ Add https://repository.mulesoft.org/nexus/content/repositories/releases[this] re
...
-                </goals>
-            </execution>
-        </executions>
-    </plugin>
-</plugins>
-----
+=== Notes
 
-In this example, the _mule-app-maven-plugin_ packages the Mule application that is deployed by the _mule-maven-plugin_.
+* You need to have a Mule Standalone Runtime Server installed locally and pointt to it with the *muleHome* property.
```
Comment (diegostrubolini) [inline mule-maven-plugin/README.adoc]: pointt -> point

File: mule-maven-plugin/README.adoc [hunk 6]
```diff
@@ -36,547 +34,178 @@ Add https://repository.mulesoft.org/nexus/content/repositories/releases[this] re
...
 
 
-== Full example
+== Deploy to Anypoint Runtime Manager
 
-In this example you will see the plugin working for Standalone deployment, configuring one applications to be deployed, two external libs to be added to the server, a domain to be deployed, and a script to be ran just before starting the server.
+You can deploy your application to a running ARM server, serverGroup or cluster.
```
Comment (diegostrubolini) [inline mule-maven-plugin/README.adoc]: should serverGroup go separately (server group)?

File: mule-maven-plugin/README.adoc [hunk 7]
```diff
@@ -36,547 +34,178 @@ Add https://repository.mulesoft.org/nexus/content/repositories/releases[this] re
...
-<5> Domain to deploy, to add your application to the domain you must configure your application manually (optional).
-<6> Optional Groovy script to be executed just before the deployment.
-<7> Use Enterprise Edition.
-<8> Use the **deploy** goal to download Mule, install it and deploy the domain and applications.
-<9> Use the **undeploy** goal to undeploy de applications and stop Mule server.
-
+=== ARM On Prem TLS errors
 
-== Deploying to a local Mule Cluster
+When trying to connect to a an On Prem installation the plugin will validate certificates for that server. If you haven't installed the server certificates in your trust store you will see an SSL error. To avoid that problem you can run the plugin in an insecure mode, this way, the security validations would be skipped. You can use the **armInsecure** tag or the **arm.insecure** system property. See the configuration example below:
```
Comment (diegostrubolini) [inline mule-maven-plugin/README.adoc]: remove the "a" in "a an On prem"

Comment (diegostrubolini) [reply mule-maven-plugin/README.adoc]: would be skipped -> will be skipped

File: mule-maven-plugin/README.adoc [hunk 8]
```diff
@@ -36,547 +34,178 @@ Add https://repository.mulesoft.org/nexus/content/repositories/releases[this] re
...
-            </goals>
-            <configuration>
-                <application>${app.2.location}</application>
-            </configuration>
-        </execution>
-    </executions>
-</plugin>
-----
+[WARNING]
+Enabling insecure connection is a very risky practice, you shouldn't use this except when you know what you are doing and your On Prem installation is isolated in a local network.
```
Comment (diegostrubolini) [inline mule-maven-plugin/README.adoc]: "Enabling an insecure connection" or "Enabling insecure connections"

File: README.adoc [hunk 9]
```diff
@@ -0,0 +1,39 @@
+= Mule Artifact Tools
+
+The project contains all the modules required to create Mule artifacts and to deploy them to the different environment of the Anypoint Platform.
```
Comment (diegostrubolini) [inline README.adoc]: environment -> environments

Comment (damiansima) [reply README.adoc]: done

File: README.adoc (hunk 1 above)
Comment (damiansima) [reply README.adoc]: done

File: README.adoc (hunk 2 above)
Comment (damiansima) [reply README.adoc]: done

File: README.adoc (hunk 3 above)
Comment (damiansima) [reply README.adoc]: done

File: mule-maven-plugin/README.adoc (hunk 4 above)
Comment (damiansima) [reply mule-maven-plugin/README.adoc]: done

File: mule-maven-plugin/README.adoc (hunk 6 above)
Comment (damiansima) [reply mule-maven-plugin/README.adoc]: done

File: mule-maven-plugin/README.adoc (hunk 7 above)
Comment (damiansima) [reply mule-maven-plugin/README.adoc]: done

File: mule-maven-plugin/README.adoc (hunk 8 above)
Comment (damiansima) [reply mule-maven-plugin/README.adoc]: done

**Pull request merged**